Android/app/src/main/java/com/example/boardgamesocial/MainApp/Fragments/UserGamesFragment.java

```diff
@@ -98,11 +107,32 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         recyclerView.setAdapter(dataClsAdapter);
 
         DataClsVM dataClsVM = DataClsVM.getInstance();
-        //TODO: Need to fetch user specific games from db
-        dataClsVM.getMediatorLiveData(RetrofitClient.getClient().getCall(Game.class, new HashMap<>()), Game.class, false)
+        dataClsVM.getMediatorLiveData(fetchUserSpecificGames(), Game.class, true)
                 .observe(getViewLifecycleOwner(), dataClsAdapter::addNewObjects);
     }
 
+    private Observable<JsonArray> fetchUserSpecificGames() {
+
+        Log.i(TAG, "Inside fetchUserSpecificGames");
+
+        gameToUserList = new ArrayList<>();
+
+        RetrofitClient.getClient().getCall(GameToUser.class, new HashMap<String, String>() {{
+            put("userId", String.valueOf(Utils.getUserId()));
+        }}).blockingSubscribe(gameToUserJson -> gameToUserList = getObjectList(gameToUserJson, GameToUser.class));
+
+        JsonArray jsonArray = new JsonArray();
+
+        gameToUserList.forEach(gameToUser -> RetrofitClient.getClient().getCall(Game.class, new HashMap<String, String>() {{
+            put("id", String.valueOf(gameToUser.getGameId()));
+        }}).blockingSubscribe(j -> {
+            if(!j.isEmpty())
+                jsonArray.add(j.get(0));
+        }));
```

Comment:
This probably should be done with chain requests, since second requests depends on the first one. You could use flatMap or make request in subscribe that should work too.

Reply:
Don’t have the idea of flatMap, would look into it but will help if you could make a commit for this part if possible or an example maybe